app/service/data_svc.py

```diff
@@ -357,25 +353,26 @@ async def _verify_abilities(self):
                 if not getattr(ability, field):
                     setattr(ability, field, 'auto-generated')
                     self.log.warning('Missing required field in ability %s: %s' % (ability.ability_id, field))
-            for payload in ability.payloads:
-                payload_name = payload
-                if self.is_uuid4(payload):
-                    payload_name, _ = self.get_service('file_svc').get_payload_name_from_uuid(payload)
-                if any(payload_name.endswith(extension) for extension in special_extensions) or \
-                        (ability.code and payload_name == ability.build_target):
-                    continue
-                _, path = await self.get_service('file_svc').find_file_path(payload_name)
-                if not path:
-                    self.log.warning('Payload referenced in %s but not found: %s' % (ability.ability_id, payload))
-                    continue
-                for cleanup_ability in [a for a in cleanup_abilities if a.executor == ability.executor]:
+            for executor in ability.executors:
+                for payload in executor.payloads:
+                    payload_name = payload
                     if self.is_uuid4(payload):
-                        decoded_command = ability.replace_cleanup(cleanup_ability.cleanup[0], '#{payload:%s}' % payload)
-                    else:
-                        decoded_command = ability.replace_cleanup(cleanup_ability.cleanup[0], payload)
-                    cleanup_command = self.encode_string(decoded_command)
-                    if cleanup_command not in ability.cleanup:
-                        ability.cleanup.append(cleanup_command)
+                        payload_name, _ = self.get_service('file_svc').get_payload_name_from_uuid(payload)
+                    if any(payload_name.endswith(extension) for extension in special_extensions) or \
+                            (executor.code and payload_name == executor.build_target):
+                        continue
+                    _, path = await self.get_service('file_svc').find_file_path(payload_name)
+                    if not path:
+                        self.log.warning('Payload referenced in %s but not found: %s' % (ability.ability_id, payload))
```

Comment:
Nit: pass the log format string arguments into the logging call (don't format if you don't have to)